### ISSUE TITLE
Not all vendor files are copied

### ISSUE THREAD

**Issue by reinos** [angular/angular-cli#1118]
I installed Font-awesome with NPM and added to the vendor list

```
module.exports = function(defaults) {
  return new Angular2App(defaults, {
    vendorNpmFiles: [
      'systemjs/dist/system-polyfills.js',
      'systemjs/dist/system.src.js',
      'zone.js/dist/*.js',
      'es6-shim/es6-shim.js',
      'reflect-metadata/*.js',
      'rxjs/**/*.js',
      '@angular/**/*.js',
      '@angular2-material/**/*.js',
      'angular2-jwt/**/*.+(js|js.map)',
      'font-awesome/fonts/**/*.*'
    ]
  });
};
```

In the folder `fonts` are 6 files
![screen shot 2016-06-17 at 15 48 12](https://cloud.githubusercontent.com/assets/633730/16152596/f1bad744-34a2-11e6-9c38-4d6667135af1.png)

but the system just copy one file `FontAwesome.otf`
![screen shot 2016-06-17 at 15 51 06](https://cloud.githubusercontent.com/assets/633730/16152691/5aad6960-34a3-11e6-95b0-355d2485790d.png)

Is there something wrong with the path or with the font name?

**Comment by RicardoVaranda** [comment]
Change:
'font-awesome/fonts/**/*.*'

To:
'font-awesome/fonts/**/*.***'
or:
'font-awesome/fonts/**/*.+(js|css|whatever)'

change the whatever to meet your needs :)

**Comment by reinos** [comment]
@RicardoVaranda that give me the same result, still just 1 file copied

`font-awesome/fonts/**/*.***` and `font-awesome/**/*.+(otf|eot|svg|ttf|woff|woff2)`

**Comment by sqwk** [comment]
I have the same problem with `moment/moment.js`and `'webrtc-adapter/out/adapter_no_edge_no_global.js`. The problem only occurs for a production build. Oddly enough `font-awesome/fonts/**/*`works.

**Comment by RicardoVaranda** [comment]
Have you tried adding two * before the .  font-awesome/**/**.**

Also what version of cli are you using? I'm on beta 5 and it's working for me just fine

**Comment by vaudy** [comment]
are you using ng build -prod? because the -prod remove js files that's not provided in polyfills

```
return new Angular2App(default, {
  vendorNpmFiles: [...],
  polyfills: [
    'system.js',
    'zone.js',
    etc..
  ]
})
```


see Angular2App.js in angular-cli/lib/brocolli/angular2-app.js for the complete list of default polyfills

**Comment by sqwk** [comment]
Yes, I was using -prod.I have followed the wiki for adding moment.js. (although there were also a couple more js files that were not copied).

Adding `moment/monent.js` to `vendorNpmFiles` worked:

    return new Angular2App(defaults, {
        vendorNpmFiles: [
            'systemjs/dist/system-polyfills.js',
            'systemjs/dist/system.src.js',
            'zone.js/dist/**/*.+(js|js.map)',
            'es6-shim/es6-shim.js',
            'reflect-metadata/**/*.+(ts|js|js.map)',
            'rxjs/**/*.+(js|js.map)',
            '@angular/**/*.+(js|js.map)',
            'moment/moment.js'
        ],
        polyfills: [
            'vendor/es6-shim/es6-shim.js',
            'vendor/reflect-metadata/Reflect.js',
            'vendor/systemjs/dist/system.src.js',
            'vendor/zone.js/dist/zone.js',
            'vendor/moment/moment.js'
        ]
    });


However, an additional `defaultExtension: 'js'` was also needed in `system-config.ts` for it to pick up the js file in -prod mode. Should I add this to the wiki?

    const packages: any = {
        'moment':{
            format: 'cjs',
            defaultExtension: 'js'
        }
    };

**Comment by BenevidesLecontes** [comment]
This issue persist? Because my css files are copied but js files not, using --prod flag.

**Comment by cironunes** [comment]
I tried to reproduce both scenarios: font-awesome and moment.

Just by adding them to the `vendorNpmFiles` list like that:

```
      'moment/moment.js',
      'font-awesome/fonts/*.*'
```

they were copied. I used the current master, which is the `beta-8` at the moment. I'll close this. Feel free to reopen if it's still an issue.

**Comment by vaudy** [comment]
the problem is when u add a custom array of polyfills, the -prod remove files not involved in this array.

**Comment by BenevidesLecontes** [comment]
@cironunes try to use bootstrap and include jquery and bootstrap.js. Serve with --prod flag and i think you'll see the issue.

**Comment by francisoud** [comment]
I think I have the same issue (see error below).

Strangly there is no `myapp/dist/vendor/@angular` directory ?!
But `angular-cli-build.js` contains the following line:
```
var Angular2App = require('angular-cli/lib/broccoli/angular2-app');

module.exports = function(defaults) {
  return new Angular2App(defaults, {
    vendorNpmFiles: [
      'systemjs/dist/system-polyfills.js',
      'systemjs/dist/system.src.js',
      'zone.js/dist/**/*.+(js|js.map)',
      'es6-shim/es6-shim.js',
      'reflect-metadata/**/*.+(ts|js|js.map)',
      'rxjs/**/*.+(js|js.map)',
      '@angular/**/*.+(js|js.map)'
    ]
  });
};

```
### Steps to reproduce

```
ng new myapp
myapp> npm install
myapp> ng build -prod
myapp> ng test --build false --watch false
```

```
Running without elevated rights. Running Ember CLI "as Administrator" increases performance significantly.
See ember-cli.com/user-guide/#windows for details.

21 07 2016 10:57:44.106:WARN [watcher]: Pattern "C:/tmp/myapp/dist/vendor/systemjs/dist/system-polyfills.js" does not match any file.
21 07 2016 10:57:44.108:WARN [watcher]: Pattern "C:/tmp/myapp/dist/vendor/zone.js/dist/async-test.js" does not match any file.
21 07 2016 10:57:44.110:WARN [watcher]: Pattern "C:/tmp/myapp/dist/vendor/zone.js/dist/fake-async-test.js" does not match any file.
21 07 2016 10:57:44.148:INFO [karma]: Karma v0.13.22 server started at http://localhost:9876/
21 07 2016 10:57:44.160:INFO [launcher]: Starting browser Chrome
21 07 2016 10:57:45.128:INFO [Chrome 51.0.2704 (Windows 7 0.0.0)]: Connected on socket /#RXhnR8uuqkB_LPgFAAAA with id 34798801
21 07 2016 10:57:45.266:WARN [web-server]: 404: /base/dist/vendor/@angular/core/testing.js
21 07 2016 10:57:45.267:WARN [web-server]: 404: /base/dist/vendor/@angular/platform-browser-dynamic/testing.js
Missing error handler on `socket`.
TypeError: (msg || "").replace is not a function
    at C:\tmp\myapp\node_modules\karma\lib\reporter.js:45:23
    at onBrowserError (C:\tmp\myapp\node_modules\karma\lib\reporters\base.js:58:60)
    at .<anonymous> (C:\tmp\myapp\node_modules\karma\lib\events.js:13:22)
    at emitTwo (events.js:106:13)
    at emit (events.js:191:7)
    at onKarmaError (C:\tmp\myapp\node_modules\karma\lib\browser.js:95:13)
    at Socket.<anonymous> (C:\tmp\myapp\node_modules\karma\lib\events.js:13:22)
    at emitOne (events.js:101:20)
    at Socket.emit (events.js:188:7)
    at Socket.onevent (C:\tmp\myapp\node_modules\karma\node_modules\socket.io\lib\socket.js:335:8)
    at Socket.onpacket (C:\tmp\myapp\node_modules\karma\node_modules\socket.io\lib\socket.js:295:12)
    at Client.ondecoded (C:\tmp\myapp\node_modules\karma\node_modules\socket.io\lib\client.js:193:14)
    at Decoder.Emitter.emit (C:\tmp\myapp\node_modules\karma\node_modules\socket.io\node_modules\socket.io-parser\node_modules\component-emitter\index.js:134:20)
    at Decoder.add (C:\tmp\myapp\node_modules\karma\node_modules\socket.io\node_modules\socket.io-parser\index.js:247:12)
    at Client.ondata (C:\tmp\myapp\node_modules\karma\node_modules\socket.io\lib\client.js:175:18)
    at emitOne (events.js:96:13)
21 07 2016 10:57:47.276:WARN [Chrome 51.0.2704 (Windows 7 0.0.0)]: Disconnected (1 times)
```

```
C:\tmp\myapp>tree dist
C:\TMP\MYAPP\DIST
├───app
└───vendor
    ├───es6-shim
    ├───reflect-metadata
    │   ├───test
    │   │   └───reflect
    │   └───typings
    ├───systemjs
    │   └───dist
    └───zone.js
        └───dist
```

### Version
```
>ng --version
angular-cli: 1.0.0-beta.8
node: 6.2.1
os: win32 x64
```

**Comment by hhsadiq** [comment]
having same issue. Did anyone able to fix it.

```
module.exports = function(defaults) {
  return new Angular2App(defaults, {
    vendorNpmFiles: [
      'systemjs/dist/system-polyfills.js',
      'systemjs/dist/system.src.js',
      'zone.js/dist/**/*.+(js|js.map)',
      'es6-shim/es6-shim.js',
      'reflect-metadata/**/*.+(ts|js|js.map)',
      'rxjs/**/*.+(js|js.map)',
      '@angular/**/*.+(js|js.map)',
      'moment/moment.js',
      'lodash/lodash.js',
      'immutable/dist/immutable.js',
      'ng2-bootstrap/**/*.js',
      'ng2-select/**/*.+(js|css)',
      'chart.js/dist/Chart.bundle.js',
      'ng2-charts/**/*.js'
    ]
  });
};```

Where as only small subset is created in vendor folder for prod build

![image](https://cloud.githubusercontent.com/assets/8002757/17068191/f2d24b6c-5068-11e6-998b-7e05e429deab.png)

**Comment by martinleopold** [comment]
Same problem: Not all vendor files are copied to `dist/vendor` when using `ng build -prod` or `ng serve -prod`. This results in the app not starting up. In the development environment it works just fine.

I have investigated the source a little, and I suspect it’s the call to `_getBundleTree` [here](https://github.com/angular/angular-cli/blob/master/lib/broccoli/angular2-app.js#L160) since it’s the only call gated by the production flag.
I think in the [`_getBundleTree` method](https://github.com/angular/angular-cli/blob/master/lib/broccoli/angular2-app.js#L422) the `vendorTree` gets left behind.
I tried adding it back in by changing [line 483](https://github.com/angular/angular-cli/blob/master/lib/broccoli/angular2-app.js#L483) from:
```ts
return BroccoliMergeTrees([nonJsTree, scriptTree, bundleTree], { overwrite: true });
```
to:
```ts
return BroccoliMergeTrees([nonJsTree, scriptTree, vendorTree, bundleTree], { overwrite: true });
```
and it seems to work. Vendor files now get copied, app builds and runs.

My understanding of broccoli is minimal, so this is pretty much guesswork at that point. Can somebody confirm, @cironunes?

**Comment by filipesilva** [comment]
Closed as obsolete following #1455. All files in `./public` should be copied now. 

There is another issue with fonts being tracked in https://github.com/angular/angular-cli/issues/1463 though.
